src/types/mapping.rs

```diff
@@ -102,18 +108,46 @@ impl PyMapping {
                 .from_owned_ptr_or_err(ffi::PyMapping_Items(self.as_ptr()))
         }
     }
+
+    /// Register a pyclass as a subclass of `collections.abc.Mapping` (from the Python standard
+    /// library). This is equvalent to `collections.abc.Mapping.register(T)` in Python.
+    /// This registration is required for a pyclass to be downcastable from `PyAny` to `PyMapping`.
+    pub fn register<T: PyTypeInfo>(py: Python<'_>) -> PyResult<()> {
+        let ty = T::type_object(py);
+        get_mapping_abc(py)?.call_method1("register", (ty,))?;
+        Ok(())
+    }
+}
+
+fn get_mapping_abc(py: Python<'_>) -> Result<&PyType, PyErr> {
+    MAPPING_ABC
+        .get_or_init(py, || {
+            Ok(py
+                .import("collections.abc")?
+                .getattr("Mapping")?
+                .downcast::<PyType>()?
+                .into_py(py))
+        })
+        .as_ref()
+        .map(|t| t.as_ref(py))
+        .map_err(|e| e.clone_ref(py))
```

Comment:
The overall approach is definitely correct, you can actually do this bit in one line:

```suggestion
        .map_or_else(|e| e.clone_ref(py), |t| t.as_ref(py))
```

Reply:
I had to add `Ok` and `Err` because `map_or_else` takes a `Result<T, E> -> U`, but otherwise this worked - thanks!